samza-core/src/main/java/org/apache/samza/runtime/RemoteApplicationRunner.java

```diff
@@ -46,25 +53,91 @@ public RemoteApplicationRunner(Config config) {
    * Run the {@link StreamApplication} on the remote cluster
    * @param app a StreamApplication
    */
-  @Override public void run(StreamApplication app) {
+  @Override
+  public void run(StreamApplication app) {
     try {
-      // 1. build stream graph
-      StreamGraph streamGraph = new StreamGraphImpl(this, config);
-      app.init(streamGraph, config);
+      // 1. initialize and plan
+      ProcessorGraph processorGraph = getExecutionPlan(app);
 
-      // 2. create the physical execution plan
-      ExecutionPlanner planner = new ExecutionPlanner(config);
-      ProcessorGraph processorGraph = planner.plan(streamGraph);
+      // 2. create the necessary streams
+      List<StreamSpec> streams = processorGraph.getIntermediateStreams().stream()
+          .map(streamEdge -> streamEdge.getStreamSpec())
+          .collect(Collectors.toList());
+      StreamManager streamManager = new StreamManager(new JavaSystemConfig(config).getSystemAdmins());
+      streamManager.createStreams(streams);
 
       // 3. submit jobs for remote execution
       processorGraph.getProcessorNodes().forEach(processor -> {
           Config processorConfig = processor.generateConfig();
-          log.info("Starting processor {} with config {}", processor.getId(), config);
+          log.info("Starting processor {} with config {}", processor.getId(), processorConfig);
           JobRunner runner = new JobRunner(processorConfig);
           runner.run(true);
         });
     } catch (Throwable t) {
-      throw new SamzaException("Fail to run application", t);
+      throw new SamzaException("Failed to run application", t);
     }
   }
+
+  @Override
+  public void kill(StreamApplication app) {
+    try {
+      ProcessorGraph processorGraph = getExecutionPlan(app);
+
+      processorGraph.getProcessorNodes().forEach(processor -> {
+          Config processorConfig = processor.generateConfig();
+          log.info("Killing processor {} with config {}", processor.getId(), processorConfig);
+          JobRunner runner = new JobRunner(processorConfig);
+          runner.kill();
+        });
+    } catch (Throwable t) {
+      throw new SamzaException("Failed to kill application", t);
+    }
+  }
+
+  @Override
+  public ApplicationStatus status(StreamApplication app) {
+    try {
+      boolean finished = false;
+      boolean unsuccessfulFinish = false;
+
+      ProcessorGraph processorGraph = getExecutionPlan(app);
+      for (ProcessorNode processor : processorGraph.getProcessorNodes()) {
+        Config processorConfig = processor.generateConfig();
+        JobRunner runner = new JobRunner(processorConfig);
+        ApplicationStatus status = runner.status();
+        log.debug("Status is {} for processor {} with config {}", new Object[]{status, processor.getId(), processorConfig});
+
+        switch (status) {
+          case Running:
+            return ApplicationStatus.Running;
+          case UnsuccessfulFinish:
+            unsuccessfulFinish = true;
```

Comment:
IIUC, probably cleaner to return ApplicationStatus.UnsuccessfulFinish immediately here. Don't need the `unsuccessfulFinish` boolean that way.